R/prepare_weo_2024_hybrid_in_ev_scenario.R

```diff
@@ -248,234 +263,98 @@ weo_2024_extract_power <- function(weo_2024_ext_data_regions_raw,
-    dplyr::full_join(
-      weo_2024_auto_tech_share,
-      by = c("scenario", "technology", "year", "region", "units")
-    )
+  iea_ev <- iea_sales_share_ev_ldv %>%
+    dplyr::left_join(iea_sales_longer_global, by = c("Year", "Scenario"))
 
-  weo_2024_automotive_with_completed_nze <-
-    weo_2024_automotive_with_share %>%
-    dplyr::mutate(
-      value_sector = dplyr::if_else(
-        .data[["scenario"]] == "NZE" & .data[["year"]] == 2030,
-        .data[["value"]] / .data[["value_share"]],
-        .data[["value_sector"]]
-      )
-    ) %>%
+  iea_ev_ice <- iea_ev %>%
     dplyr::mutate(
-      value_sector = zoo::na.locf(.data[["value_sector"]]),
-      .by = c(
-        "scenario",
-        "year",
-        "units"
-      )
-    ) %>%
-    dplyr::mutate(
-      value = dplyr::if_else(
-        is.na(.data[["value"]]),
-        .data[["value_sector"]] * .data[["value_share"]],
-        .data[["value"]]
-      ),
-      value_share = NULL
+      ICE = Electric * (100 / `Sales share` - 1)
     )
 
-  weo_2024_automotive_nze_with_ice <-
-    weo_2024_automotive_with_completed_nze %>%
-    # Big Assumption:
-    # By WEO standards, "Total" indicates anything PHEV, BEV and Fuel Cell
-    # By AI standards, this means the "inverse" of Total = "ICE"
-    dplyr::filter(.data[["year"]] == "2030", .data[["scenario"]] == "NZE") %>%
-    tidyr::pivot_wider(
-      names_from = "technology",
-      values_from = "value"
-    ) %>%
-    dplyr::mutate(
-      ICE = .data[["value_sector"]] - (.data[["BEV"]] + .data[["PHEV"]] + .data[["FuelCell"]])
-    ) %>%
+  iea_auto_end_loop <- iea_ev_ice %>%
+    dplyr::select(-c("Unit.x", "Source.x", "Source.y", "Sales share", "Technology", `Vehicle Type`)) %>%
     tidyr::pivot_longer(
-      cols = c("Total", "BEV", "PHEV", "FuelCell", "ICE"),
-      names_to = "technology",
-      names_prefix = "value_",
-      values_to = "value"
-    )
-
-  weo_2024_automotive <-
-    dplyr::bind_rows(
-      dplyr::filter(weo_2024_automotive_with_share, !(.data[["scenario"]] == "NZE" & .data[["year"]] == 2030)),
-      weo_2024_automotive_nze_with_ice
+      cols = c("Electric", "ICE"),
+      names_to = "Technology",
+      values_to = "Value"
     ) %>%
+    dplyr::rename(Units = "Unit.y") %>%
+    dplyr::mutate(Units = "# (in million)")
+
+  weo_2024_automotive <- iea_auto_end_loop %>%
     dplyr::mutate(
-      source = "WEO2024",
-      scenario_geography = "Global",
       sector = "Automotive",
-      variable = "Sales",
-      value = .data[["value"]] / 1000000,
-      units = "# (in million)",
-      indicator = "Sales"
-    ) %>%
-    dplyr::filter(
-      .data[["technology"]] != c("Total"),
-    ) %>%
-    dplyr::summarize(value = sum(.data[["value"]]), .by = -c("value")) %>%
+      indicator = "Sales",
+      Technology = dplyr::case_when(
+        Technology == "BEV" ~ "Electric",
+        Technology == "PHEV" ~ "Hybrid",
+        TRUE ~ Technology
+      )) %>%
     dplyr::select(
```

Comment:
style:
generally to make it easier to see what exactly is happening where, I would suggest not to rename a variable in a inside a `select()` call, but rather add a few more lines and do the renaming in `rename()`. Not a required change, just a suggestion